tests/test_hub.py

```diff
@@ -0,0 +1,52 @@
+from pathlib import Path
+
+import pytest
+from huggingface_hub.utils import LocalEntryNotFoundError
+
+from vllm_tgis_adapter.tgis_utils.hub import (
+    convert_files,
+    download_weights,
+    weight_files,
+    weight_hub_files,
+)
+
+
+def test_convert_files():
+    model_id = "bigscience/bloom-560m"
+    local_pt_files = download_weights(model_id, extension=".bin")
+    local_pt_files = [Path(p) for p in local_pt_files]
+    local_st_files = [
+        p.parent / f"{p.stem.removeprefix('pytorch_')}.safetensors"
+        for p in local_pt_files
+    ]
+    convert_files(local_pt_files, local_st_files, discard_names=[])
+
+    found_st_files = weight_files(model_id)
+
+    assert all(str(p) in found_st_files for p in local_st_files)
+
+
+def test_weight_hub_files():
+    filenames = weight_hub_files("bigscience/bloom-560m")
+    assert filenames == ["model.safetensors"]
+
+
+def test_weight_hub_files_llm():
+    filenames = weight_hub_files("bigscience/bloom")
+    assert filenames == [f"model_{i:05d}-of-00072.safetensors" for i in range(1, 73)]
+
+
+def test_weight_hub_files_empty():
+    filenames = weight_hub_files("bigscience/bloom", ".errors")
+    assert filenames == []
+
+
+def test_download_weights():
+    files = download_weights("bigscience/bloom-560m")
+    local_files = weight_files("bigscience/bloom-560m")
+    assert files == local_files
```

Comment:
This test tries to download 350GB worth of files, That's way too much to run in a CI pipeline, I'd prefer to either use something way smaller, or make all tests which download lots of data from hf hub hidden behind a custom marker:

https://docs.pytest.org/en/latest/example/markers.html

Reply:
Should be possible to use a tiny model to test out this scenario

Reply:
It would be good to have unit tests for the functions that emulate the behavior somehow (can be tackled in subsequent PRs) - worth looking into the custom marker

Reply:
Went ahead and added a custom marker `large` to this test, although it seems a bit generic. If there's a preferred name just let me know!